### ISSUE TITLE
CA1801: Is not able to correctly detect used parameter if it is used only to pass it to base constructor.

### ISSUE THREAD

**Issue by mchaloupka** [dotnet/roslyn-analyzers#1264]
#### Analyzer package

Example: Microsoft.CodeAnalysis.FxCopAnalyzers

#### Repro steps

Try the following code:
```
public CommunicationTimeoutException(TimeSpan timeOut)
     : base($"Waiting for agent response timed out after {timeOut}.")
{
            
}
```

#### Expected behavior
The static analysis should go without error.

#### Actual behavior
The following error is returned:
CA1801: Parameter timeOut of method .ctor is never used. Remove the parameter or use it in the method body.

**Comment by mavasani** [comment]
This is likely an IOperation bug. I will investigate later today

**Comment by 333fred** [comment]
@mchaloupka, what version of the analyzers are you using? We've fixed a number of bugs in later versions, this is probably https://github.com/dotnet/roslyn-analyzers/issues/1061.

**Comment by mchaloupka** [comment]
The used version is 2.0.0-beta2. I will try to update it to a newer version and check whether it is fixed or not.

**Comment by 333fred** [comment]
@mchaloupka, yeah, that's probably the cause then. We're still actively working on the API the beta versions of the analyzers use, and the 2.0.0-beta1 version wasn't able to look at most C# 7.0 features, so the interpolated string is skipped. You should be able to use the newer analyzers if you also upgrade your compiler, https://www.nuget.org/packages/Microsoft.Net.Compilers/.

**Comment by mchaloupka** [comment]
@333fred I have tried to add dependency on `Microsoft.Net.Compilers` version `2.3.1` and upgrade `Microsoft.CodeAnalzsis.FxCopAnalyzers` to version `2.3.0-beta1` but I was not able to get them even working. I have got the following warning:

>Microsoft.CodeQuality.CSharp.Analyzers.ApiDesignGuidelines.CSharpDefineAccessorsForAttributeArgumentsAnalyzer cannot be created from C:\Users\***\.nuget\packages\microsoft.codequality.analyzers\2.3.0-beta1\analyzers\dotnet\cs\Microsoft.CodeQuality.CSharp.Analyzers.dll: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=2.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.

And many other. I have found this warning and it sounded to me like a known issue which will be fixed in VS 2017.3 with some workaround (like putting the assemblies into GAC - that does not sound acceptable to me). So, if you believe that it is working then close this issue and I will wait till I will get some newer version working.

**Comment by 333fred** [comment]
Alright then. If you still have this issue after 2017.3 comes out, feel free to open up a new issue.
